docs/core/diagnostics/distributed-tracing-builtin-activities.md

```diff
@@ -0,0 +1,151 @@
+
+### HTTP client request: wait for connection (experimental)
+
+This activity is a child of an *HTTP client request* activity. It represents the time interval the corresponding request is waiting for an available connection in the request queue. If a free connection is available in the pool when the request comes in, no *wait for connection* activity will be created. Note that *wait for connection* does not represent the actual connection establishment; that is modeled by the *HTTP connection setup* activity.
+
+| Availability | <xref:System.Diagnostics.ActivitySource> name | <xref:System.Diagnostics.Activity.OperationName> | <xref:System.Diagnostics.Activity.DisplayName> |
+|---|---|---|---|
+| .NET 9+ | `Experimental.System.Net.Http.Connections` | `Experimental.System.Net.Http.Connections.WaitForConnection` | `HTTP wait_for_connection {address}:{port}`  |
+
+> [!TIP]
+> The time it takes to get a connection from the pool is also reported by the [`http.client.request.time_in_queue`](built-in-metrics-system-net.md#metric-httpclientrequesttime_in_queue) metric.
+
+> [!WARNING]
+> This activity is experimental. It might be altered or deleted in a future version.
+
+#### Attributes (tags)
+
+| Attribute  | Type | Description  | Examples  | Presence |
+|---|---|---|---|---|
+| `error.type` | `string` | The connection failure reason: one of the [HTTP request errors](xref:System.Net.Http.HttpRequestError) in snake_case, or a full exception type. | `System.OperationCanceledException`; `name_resolution_error`; `secure_connection_error` | If the connection attempt fails. |
+
+### HTTP connection setup (experimental)
+
+This activity describes the establishment of an HTTP connection. Typically, this includes the the time it takes to resolve the DNS, establish the socket connection, and perform the TLS handshake.
+
+| Availability | <xref:System.Diagnostics.ActivitySource> name | <xref:System.Diagnostics.Activity.OperationName> | <xref:System.Diagnostics.Activity.DisplayName> |
+|---|---|---|---|
+| .NET 9+ | `Experimental.System.Net.Http.Connections` | `Experimental.System.Net.Http.Connections.ConnectionSetup` | `HTTP connection_setup {address}:{port}`  |
+
+There is no parent-child relationship between the *HTTP client request* and the *HTTP connection setup* activities; the latter will always be a root activity, living in a separate trace. However, if the connection attempt represented by *HTTP connection setup* results in a succesful HTTP connection, and that connection is picked up by a request to serve it, the instrumentation adds an <xref:System.Diagnostics.ActivityLink> to the *HTTP client request* activity pointing to the *HTTP connection setup* activity. That is, each request is linked to the connection that served the request.
+
+> [!NOTE]
+> If *HTTP connection setup* fails, it won't be linked to any *HTTP client request*.
```

Comment:
Your docs are correct, but should we consider changing this if we're failing an ininiating request due to the connection failure?

Reply:
Changing that behavior would be a huge breaking change and would render a bunch of `HttpRequestError`s only observable via telemetry API-s. Another approach would be to add another *Injected Connection* link that would be also available for failed connections and match the exception behavior, but that might expose implementation details we may want to change later.

Reply:
We actually considered this, but abandoned chasing the idea because of time pressure. Noted it in https://github.com/dotnet/runtime/issues/110593.